### PR TITLE
Tyler us25 26

### PR DESCRIPTION
**Pull request by tylerpporter**
- User must be logged in to checkout
- Updated tests to reflect login requirements 
- Orders have default status of 'pending'

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,35 +1,51 @@
...
+  def successful_redirect
+    session.delete(:cart)
+    flash[:notice] = "Your order has been created"
+    redirect_to "/profile/orders"
+  end
+
+  def success(order)
+    create_item_orders(order)
+    successful_redirect
+  end
```
Comment (Shock253) [inline app/controllers/orders_controller.rb:34]: this is the only place where successful_redirect is called, I'm not sure it's necessary. success could have this functionality and still be a small enough chunk of code

**Review by Shock253 — COMMENTED** [review]
Good for the most part, maybe a minor refactor in orders controller

**Review by Shock253 — APPROVED** [review]

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
-          })
-      end
-      session.delete(:cart)
-      redirect_to "/orders/#{order.id}"
-    else
-      flash[:notice] = "Please complete address form to create an order."
-      render :new
-    end
+    order = current_user.orders.create(order_params)
+    order.save ? success(order) : failure
```
Comment (tony-gomes) [inline app/controllers/orders_controller.rb:16]: great ternary here...

**Review by cmpprg — APPROVED** [review]
All issues resolved in person.

**Review by tony-gomes — APPROVED** [review]
looking good...

**Pull request merged**